build_and_test.sh

```diff
@@ -0,0 +1,126 @@
+#!/bin/bash
+#
+# Copyright 2017-present The kokoro iOS runner Authors. All Rights Reserved.
+# 
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# 
+# http://www.apache.org/licenses/LICENSE-2.0
+# 
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Build script for kokoro.
+#
+# This script will clean, build, and run tests against each installation of Xcode 8 and 9 available
+# on the machine.
+#
+# Arguments:
+#   1. Path to xcodeproj
+#   2. Scheme to build and test.
+#   3. grep filter for iPhone simulator types [optional].
+#
+# Example usage:
+#   build_and_test "MotionInterchange/MotionInterchange.xcodeproj" MotionInterchange "iPhone SE"
+
+# Fail on any error.
+set -e
+
+script_version="v1.0.0"
+echo "build_and_test version $script_version"
+
+project="$1"
+scheme="$2"
+target_simulator_filter="$3"
+
+# Dependencies
+
+if [ -z "$KOKORO_GITHUB_PULL_REQUEST_NUMBER" ]; then
+  : # Local run - nothing to do.
+else
+  gem install xcpretty --no-rdoc --no-ri --no-document --quiet
+
+  # Move into our cloned repo
+  cd github/repo
+fi
+
+# Runs a clean build + tests using xcodebuild on the active Xcode installation.
+run() {
```

Comment:
Usually you put your functions up as high as possible in the file.

Reply:
Done.